src/NLPModels.jl

```diff
@@ -23,9 +23,18 @@ const LAGRANGIAN_HESSIAN = raw"""
 with `σ = obj_weight`
 """
 
-# Base type for an optimization model.
+"""
+    AbstractNLPModel
+
+Base type for an optimization model.
+"""
 abstract type AbstractNLPModel end
 
+"""
+    AbstractNLSModel <: AbstractNLPModel
+
+Base type for nonlinear least-squares models.
```

Comment:
This one is plural and the one above is singular. Let's make them both plural, or both singular.